axolotl-web/src/components/Message.vue

```diff
@@ -1,96 +1,71 @@
 <template>
-  <div
-    :key="message.ID"
-    :class="{
-      'col-12': true,
-      'message-container': true,
-      outgoing: message.Outgoing,
-      sent: message.IsSent && message.Outgoing,
-      read: message.IsRead && message.Outgoing,
-      delivered: message.Receipt && message.Outgoing,
-      incoming: !message.Outgoing,
-      status:
-        (message.Flags > 0 &&
-          message.Flags !== 11 &&
-          message.Flags !== 13 &&
-          message.Flags !== 14) ||
-        message.StatusMessage ||
-        (message.Attachment &&
-          message.Attachment.includes('null') &&
-          message.Message === ''),
-      hidden: message.Flags === 18,
-      error: message.SentAt === 0 || message.SendingError,
-      'group-message':
-        isGroup && (message.Flags === 0 || message.Flags === 12 || message.Flags === 13),
-    }"
-  >
-    <div
-      v-if="
-        !message.Outgoing &&
-          isGroup &&
-          (message.Flags === 0 || message.Flags === 12 || message.Flags === 13)
-      "
-      class="avatar"
-    >
+  <div :key="message.ID" :class="{
+    'col-12': true,
+    'message-container': true,
+    outgoing: message.is_outgoing,
+    sent: message.IsSent && message.is_outgoing,
+    read: message.IsRead && message.is_outgoing,
+    delivered: message.Receipt && message.is_outgoing,
+    incoming: !message.is_outgoing,
+    status:
+      (message.Flags > 0 &&
+        message.Flags !== 11 &&
+        message.Flags !== 13 &&
+        message.Flags !== 14) ||
+      message.StatusMessage ||
+      (message.Attachment &&
+        message.Attachment.includes('null') &&
+        message.Message === ''),
+    hidden: message.Flags === 18,
+    error: message.timestamp === 0 || message.SendingError,
+    'group-message':
+      isGroup && (message.Flags === 0 || message.Flags === 12 || message.Flags === 13),
+  }">
```

Comment:
There is a lot of logic here for setting the right css classes. Is there a cleaner way to make this?
And can we move the logic into funktions, if its more than one logic operation?

Reply:
Same for here, the Flags field doesn't exist anymore but we have to add something like this again.

Reply:
And i agree to move it into functions